### PR TITLE
 testdata: add test cases for extract strings

### PR DESCRIPTION
**Pull request by mewmew**
Current test cases includes:
* ASCII
* UTF-8 in a few different languages
* UTF-16 with little and big endian
* Null-terminated strings

We can extend these to also include UTF-32 and
Big-5 among others.

### CODE REVIEW

File: testdata/strings/gen.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	"log"
+	"unicode/utf16"
+)
+
+const (
+	// ref: https://en.wikipedia.org/wiki/The_quick_brown_fox_jumps_over_the_lazy_dog
+	refEN = "The quick brown fox jumps over the lazy dog"
+	// ref: https://ja.wikipedia.org/wiki/The_quick_brown_fox_jumps_over_the_lazy_dog
+	//
+	// Quick brown eagle :)
```
Comment (karlek) [inline testdata/strings/gen.go:20]: Wow, google translate is wrong. `狐` is fox: https://jisho.org/search/%E7%8B%90

File: testdata/strings/gen.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		panic(err)
+	}
+	mustWriteFile(path, buf.Bytes())
+}
+
+func genNullTerm(path, s string) {
+	mustWriteFile(path, []byte(s+"\x00"))
+}
+
+func mustWriteFile(path string, data []byte) {
```
Comment (karlek) [inline testdata/strings/gen.go:90]: Will the generated files be ignored by some path in our `.gitignore`? Otherwise, add one line for that.

**Review by karlek — COMMENTED** [review]
Good, just a few comments.

Nice with some test cases before starting with something more complex than ascii

File: testdata/strings/gen.go (hunk 1 above)
Comment (mewmew) [reply testdata/strings/gen.go:20]: Haha, yeah. I guessed so, that's why I felt I had to include it :)

/Pink flodhäst signing off to blue eagle

File: testdata/strings/gen.go (hunk 2 above)
Comment (mewmew) [reply testdata/strings/gen.go:90]: I added a `.gitignore` in the `testdata/strings` directory for `*.bin`.

Comment (karlek) [reply testdata/strings/gen.go:90]: Missed that line!

**Pull request merged**